src/main/scala/ai/datahunters/md/processor/FlattenMetadataTags.scala

```diff
@@ -32,11 +33,19 @@ case class FlattenMetadataTags(colPrefix: String,
 
     val flattenDF = inputDF.withColumn(MetadataCol, selectMetadataTagsUDF(col(MetadataCol)))
       .select(columns.head, columns.tail:_*)
-    val transformedDF = if (includeMetadataContent) {
-      flattenDF.withColumn(MetadataContentCol, concat_ws(" ", buildConcatTagList(selectedTags, colPrefix):_*))
+    val arrayFieldsToFlatten = selectArrayColumns(flattenDF)
+    val afterArraysFlatteningDF = if (flattenArrays) {
+      arrayFieldsToFlatten
+        .foldLeft(flattenDF)((previousDF, colName) => previousDF.withColumn(colName, concat_ws(",", col(colName))))
     } else {
       flattenDF
     }
+    val transformedDF = if (includeMetadataContent) {
+      afterArraysFlatteningDF.withColumn(MetadataContentCol, concat_ws(" ", buildConcatTagList(selectedTags, colPrefix):_*))
```

Comment:
I would prefer to add delimiters into constants, but for sure it will extend lines. Please consider.